### PR TITLE
[netatmo] Document extensible channels

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@jlaur @kaikreuzer : as this is only documentation and a feature which is not yet documented, I suggest to merge this PR before 3.3 is released.

@clinique : if my interpretation of the code is correct, there is no "CO2 Measurement" extensible channel type for the Healthy Home Coach. I don't know if this is deliberate and expected, or if it is a miss.

**Comment by clinique** [comment]
> @clinique : if my interpretation of the code is correct, there is no "CO2 Measurement" extensible channel type for the Healthy Home Coach. I don't know if this is deliberate and expected, or if it is a miss.

You're correct, this is a miss. (BTW there is no official documentation in API doc of measurement for healthy home coach, I discovered that it worked like weather station)

**Review by clinique — APPROVED** [review]

File: bundles/org.openhab.binding.netatmo/README.md [hunk 1]
```diff
@@ -264,6 +299,12 @@ All these channels are read only.
 
 All these channels are read only.
 
+**Extensible channels for the rain guage:**
+
+| Channel Type         | Item Type            | Description                       | Channel parameters                                 |
+|----------------------|----------------------|-----------------------------------|----------------------------------------------------|
+| sum_rain-measurement | Number:Length        | Summing rain measurement          | period (30min, 1hour, 3hours, 1day, 1week, 1month) |
```
Comment (jlaur) [inline bundles/org.openhab.binding.netatmo/README.md:306]: Is it really **sum_rain** with underscore? Just checking, as it seems inconsistent with convention used.

Comment (lolodomo) [reply bundles/org.openhab.binding.netatmo/README.md:306]: Yes.
https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/data/NetatmoConstants.java#L87

**Comment by lolodomo** [comment]
> > @clinique : if my interpretation of the code is correct, there is no "CO2 Measurement" extensible channel type for the Healthy Home Coach. I don't know if this is deliberate and expected, or if it is a miss.
> 
> You're correct, this is a miss. (BTW there is no official documentation in API doc of measurement for healthy home coach, I discovered that it worked like weather station)

Ok, I let you create a potential PR to fix that miss in OH 3.4.

File: bundles/org.openhab.binding.netatmo/README.md (hunk 1 above)
Comment (jlaur) [reply bundles/org.openhab.binding.netatmo/README.md:306]: Now wondering if this is by mistake, and if we should change it to "sum-rain" for consistency? In 3.3 all channels are renamed anyway, so last chance to fix this without further breaking. @clinique - is it by accident, or by purpose that underscore is used here?

Comment (clinique) [reply bundles/org.openhab.binding.netatmo/README.md:306]: Yes, it is not very nice but by design because I use Netatmo API keyword

**Comment by lolodomo** [comment]
@clinique : I don't know how you discovered this feature (extensible channel types). I am even not sure if this is really documented.
The method "withExtensibleChannelTypeIds" is used only by 2 bindings: the Netatmo binding and the MQTT generic binding.

**Comment by clinique** [comment]
I think, IIRW that it has been introduced with gps tracker binding. Maybe is it used dynamically only by mqtt and Netatmo, but statically by many more (I also used it in openuv binding for the safe exposure channel)

**Comment by lolodomo** [comment]
> but statically by many more (I also used it in openuv binding for the safe exposure channel)

Ah ok this is the `extensible=` in XML description.

**Comment by clinique** [comment]
Correct

**Pull request merged**